packages/vue/src/components/FSortFilterDataset/FSortFilterDataset.vue

```diff
@@ -1,3 +1,236 @@
+<script setup lang="ts" generic="T">
+import { type PropType, type Ref, computed, nextTick, onMounted, provide, ref, useTemplateRef, watch } from "vue";
+import { focus, debounce, alertScreenReader, TranslationService } from "@fkui/logic";
+import { IFlex, IFlexItem } from "../../internal-components/IFlex";
+import { FSelectField } from "../FSelectField";
+import { FIcon } from "../FIcon";
+import { FTextField } from "../FTextField";
+import { useTranslate } from "../../plugins";
+import { FSortFilterDatasetSortCallback, FSortFilterDatasetMountCallback } from "./FSortFilterDatasetInterface";
+import { sort } from "./FSortFilterSorter";
+import { filter } from "./FSortFilterFilter";
+import { type SortOrder } from "./sort-order";
+
+const $t = useTranslate();
+
+const searchString = ref("");
+const sortAttribute = ref<SortOrder>({ attribute: "", name: "", ascendingName: "", ascending: false, id: 0 });
+const sortFilterResult = ref<T[]>([]) as Ref<T[]>;
```

Comment:
Måste man casta här?

Reply:
Här får gärna någon annan rycka in och förklara men typningen blir en helt annan typ `Ref<UnwrapRefSimple<T>[], T[] | UnwrapRefSimple<T>[]>` när man använder `[]` som default-värde. Vet inte om det är en bug i vue eller hur det är tänkt men `sortFilterResult.value` fungerar inte att använda till något som behöver `T[]`.